source/ucumLhcUtils.js

```diff
@@ -145,7 +145,251 @@ export class UcumLhcUtils {
+    }
+    
+    if (returnObj.status !== 'error') {
+      try {
+        let fromUnit = null;
+
+        let parseResp = this.getSpecifiedUnit(fromUnitCode, 'convert', suggest);
+        fromUnit = parseResp['unit'];
+        if (parseResp['retMsg'])
+          returnObj.msg = returnObj.msg.concat(parseResp['retMsg']);
+        if (parseResp['suggestions']) {
+          returnObj.suggestions = {};
+          returnObj.suggestions.from = parseResp['suggestions'];
+        }
+        if (!fromUnit) {
+          returnObj.msg.push(`Unable to find a unit for ${fromUnitCode}, ` +
+            `so no conversion could be performed.`);
+        }
+
+        let toUnit = null;
+        parseResp = this.getSpecifiedUnit(toUnitCode, 'convert', suggest);
+        toUnit = parseResp['unit'];
+        if (parseResp['retMsg'])
+          returnObj.msg = returnObj.msg.concat(parseResp['retMsg']);
+        if (parseResp['suggestions']) {
+          if (!returnObj.suggestions)
+            returnObj.suggestions = {};
+          returnObj.suggestions.to = parseResp['suggestions'];
+        }
+        if (!toUnit) {
+          returnObj.msg.push(`Unable to find a unit for ${toUnitCode}, ` +
+                                `so no conversion could be performed.`);
+        }
+
+        if (fromUnit && toUnit) {
+          try {
+            // if no molecular weight or valence was specified perform a normal conversion
+            if (!molecularWeight && !valence) {
+              returnObj.toVal = toUnit.convertFrom(fromVal, fromUnit);
+            } 
+            // if a valence was specified, but not a molecular weight, assume a mole <-> equivalent conversion
+            else if (valence && !molecularWeight) {
+              if (fromUnit.equivalentExp_ !== 0 && toUnit.equivalentExp_ !== 0) {
+                throw new Error('A valence was specified ' + 'but a mole <-> equivalent conversion cannot be executed for two ' + 'equivalent-based units.  No conversion was attempted.');
+              }
+              if (fromUnit.equivalentExp_ === 0 && toUnit.equivalentExp_ === 0) {
+                throw new Error('A valence was specified ' + 'but a mole <-> equivalent conversion cannot be executed when ' + 'neither unit is equivalent-based.  No conversion was attempted.');
+              }
+              // if from is equivalent and to is moles, assume eq to mol conversion
+              if (fromUnit.equivalentExp_ !== 0 && toUnit.moleExp_ !== 0) {
+                returnObj['toVal'] = fromUnit.convertEqToMol(fromVal, toUnit, valence);
+              } 
+              // else if from is moles and to is equivalent, assume mol to eq conversion
+              else if (fromUnit.moleExp_ !== 0 && toUnit.equivalentExp_ !== 0) {
+                returnObj['toVal'] = fromUnit.convertMolToEq(fromVal, toUnit, valence);
+              }
+            } 
+            // if a valence and molecular weight was specified, assume a mass <-> equivalent conversion
```

Comment:
I'm not sure this assumption is correct. A valence and molecular weight are both needed to handle a mass <-> eq conversion, but that doesn't mean that every user who provides both means to do a mass <-> eq conversion. 

It is possible they are actually trying to do a mol <-> eq conversion, but are just providing molecular weight as excess information that won't be used. This code will fail to do a mol <-> eq conversion and assume incorrectly that because a molecular weight was passed, the user wants a mass <-> eq

Reply:
Added handling, resolved. Will add tests